# Determination - Deterministic rendering environment for white-axe's music
# Copyright (C) 2024 Liu Hao <whiteaxe@tuta.io>
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.

# This patch fixes a compilation error if you include
# <faust/dsp/interpreter-dsp-c.h> in a C program.

--- a/architecture/faust/dsp/libfaust-box-c.h
+++ b/architecture/faust/dsp/libfaust-box-c.h
@@ -434,9 +434,9 @@ extern "C"
      *
      * @return the result box of op(x,y).
      */
-    LIBFAUST_API Box CboxBinOp(SOperator op);
+    LIBFAUST_API Box CboxBinOp(enum SOperator op);
     
-    LIBFAUST_API Box CboxBinOpAux(SOperator op, Box b1, Box b2);
+    LIBFAUST_API Box CboxBinOpAux(enum SOperator op, Box b1, Box b2);
     
     /**
      * Specific binary mathematical functions.
